docs/architecture/adr-027-deterministic-protobuf-serialization.md

```diff
@@ -45,11 +46,25 @@ with the following additions:
    must be omitted
 4. `repeated` fields of scalar numeric types must use
    [packed encoding](https://developers.google.com/protocol-buffers/docs/encoding#packed)
-   by default.
-5. Variant encoding of integers must not be longer than needed.
+5. Varint encoding must not be longer than needed:
+    * No trailing zero bytes (in little endian, i.e. no leading zeroes in big
+      endian). Per rule 3 above, the default value of `0` must be omitted, so
+      this rule does not apply in such cases.
+    * The maximum value for a varint is `FF FF FF FF FF FF FF FF FF 01`.
+      In other words, when decoded, the highest 6 bits of the 70-bit unsigned
+      integer must be `0`. (10-byte varints are 10 groups of 7 bits, i.e.
+      70 bits, of which only the lowest 70-6=64 are useful.)
+    * The maximum value for 32-bit values in varint encoding is `FF FF FF FF 0F`
+      with one exception (below). In other words, when decoded, the highest 38
+      bits of the 70-bit unsigned integer must be `0`.
+        * The one exception to the above is _negative_ `int32`, which is
+          encoded using the full 10 bytes for sign extension.
```

Comment:
Could you add a reference as a footer?

https://developers.google.com/protocol-buffers/docs/encoding#signed_integers

Reply:
Fixed in ec915f3.